rpcs3/Emu/RSX/NV47/HW/nv47_sync.hpp

```diff
@@ -27,6 +27,9 @@ namespace rsx
 					return;
 				}
 
+				// Syncronization point, may be associated with memory changes without actually changing addresses
+				RSX(ctx)->m_graphics_state |= rsx::pipeline_state::fragment_program_ucode_dirty;

```

Comment:
Move to caller or obey flush flags. Not all synchronization points mean that CPU is fully synced. e.g Texture read semaphore without SRM should ignore slow activity.
Only FlushPipe does a full sync, FlushDMA does not.